Store images db

jamiander opened the pull request:
Added image functionality for Garden List and Garden View.  I did need to make changes to the Garden class.

Commit:
will store image in uploads directory

Commit:
stores images and pairs with garden_id

Commit:
upload images, display images on Water Gardens page and on individual garden pages.

jamiander committed:
Merge branch 'master' into store_images_db

scottpenneman commented on src/main/java/org/launchcode/water_garden_tour/controllers/GardenController.java:
```diff
@@ -158,7 +183,6 @@ public String updateGarden(Model model,
         return "gardens/admin-list";
     }
 
```
I'm excited to see this!
Right now I'm getting an error on line 173 (`gardenToUpdate.setImage(image);`) when trying to compile: 
`error: incompatible types: String cannot be converted to byte[]`
Are you running into that or know a solution?

jamiander replied:
You're too quick for me!  I was still in the process of fixing/adjusting to handle the new image type and uploads with Ryan's changes.  Should work now.  Thanks!

Commit:
made changes to GardenController and update.html to handle image uploads.

Commit:
fixed 403 error

jamiander committed:
Merge branch 'master' into store_images_db

jamiander commented:
I just updated with Heather's changes.  Fixed the 403 error.  All is working!

hstrodtman reviewed (approved):
Good job!

MulesKC commented on src/main/resources/templates/fragments.html:
```diff
@@ -25,8 +25,10 @@
-                <a class="nav-link" th:href="@{/gardens/list}" sec:authorize="isAuthenticated()">Water Gardens</a>
+                <a class="nav-link list-link" th:href="@{/gardens/list}" sec:authorize="isAuthenticated()">Water Gardens</a>
                 <a class="nav-link map-link" th:href="@{/map}" sec:authorize="isAuthenticated()">Map</a>
+                <a class="nav-link" th:href="@{/features/}" sec:authorize="isAuthenticated()">Features</a>
```
I had removed these in my last pull request since they could be accessed through the /admin page now. Did you want to keep them in the navbar and hide from normal users? I wouldn't have a strong preference either way. We can discuss in class later if you want!

MulesKC reviewed (commented):
The image uploading is awesome! I hope that it wasn't too painful to merge with my work! 

I did find a small bug: if you have a garden with a picture already uploaded, and you go to update the garden's description (or something other than the photo), you lose the photo. 

I added a line comment/question about the navbar as well. 

Besides that everything looks and works great!

jamiander committed:
Added conditional for updating image

Garden image will not be over-written if an image is not uploaded during garden/update.

KBeas21 reviewed (approved):
Great caches and conversation guys!

jamiander committed:
Removed Features and Owners from fragments
